src/app/services/table.service.ts

```diff
@@ -1,22 +1,46 @@
 import {Injectable} from '@angular/core';
 import {Dimension} from '../models/table-models/dimension';
+import {DataTableState} from "../models/table-models/data-table-state";
+import {Order} from "../models/table-models/order";
 
 @Injectable()
 export class TableService {
 
   private _candidates: Dimension[];
   private _rows: Dimension[];
   private _columns: Dimension[];
+  private _sorting : Map<string, Order>;
```

Comment:
not sure I understand here, how can we tell which are the pivot row and column, which are needed for sending API call to the backend ?

Reply:
it can be distinguished based on what was saved as a row and what was saved as a column

Reply:
what's the key of this map? is it 'rows' and 'cols'?